api/src/main/java/com/github/retrooper/packetevents/wrapper/play/client/WrapperPlayClientEditBook.java

```diff
@@ -40,9 +41,15 @@ public class WrapperPlayClientEditBook extends PacketWrapper<WrapperPlayClientEd
     private static final int MAX_PAGES = 100;
 
     private int slot;
-    private List<String> pages;
+
+    // >= 1.17.1
+    private @Nullable List<String> pages;
     private @Nullable String title;
 
+    // < 1.17.1
+    private @Nullable ItemStack itemStack;
+    private @Nullable Boolean signing;
```

Comment:
```suggestion
    /**
     * Added with 1.17.1 - this will only be null for versions older than 1.17.1
     */
    private @Nullable List<String> pages;
    /**
     * Added with 1.17.1
     */
    private @Nullable String title;

    /**
     * Removed with 1.17.1
     */
    private @Nullable ItemStack itemStack;
    /**
     * Removed with 1.17.1
     */
    private @Nullable Boolean signing;
```